sw/host/hsmtool/src/util/wrap.rs

```diff
@@ -29,6 +30,28 @@ pub enum Wrap {
     AesKeyWrapPad,
     RsaPkcs,
     RsaPkcsOaep,
+    VendorThalesAesKwp,
+}
+
+/// The wrapping mechanism to use for private keys.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Display, EnumString, ValueEnum,
+)]
+#[strum(ascii_case_insensitive)]
+pub enum WrapPrivateKey {
+    AesKeyWrap,
+    AesKeyWrapPad,
+    VendorThalesAesKwp,
+}
+
+impl From<&WrapPrivateKey> for Wrap {
+    fn from(wrap: &WrapPrivateKey) -> Self {
+        match wrap {
+            WrapPrivateKey::AesKeyWrap => Wrap::AesKeyWrap,
+            WrapPrivateKey::AesKeyWrapPad => Wrap::AesKeyWrapPad,
+            WrapPrivateKey::VendorThalesAesKwp => Wrap::VendorThalesAesKwp,
+        }
+    }
```

Comment:
Since `WrapPrivateKey` is a simple enum and is `Copy`, you don't need to take a reference in this `From` conversion.  I think that might be able to simplify your `unwrap_mechansim` conversions:
```
        let wrapper: Wrap = self
            .unwrap_mechanism
            .as_ref()          // Should no longer need this `as_ref` call.
            .ok_or(anyhow!(
                "unwrap_mechanism is required when wrap is specified"
            ))?
            .into();
```

Reply:
Thanks for explaining this. Makes sense. Fixed.